In [2]:
import csv
import sqlite3

### Connect to Sqlite3 
### Create a Database called Walmart_Coffee_Listings

In [3]:
connects = sqlite3.connect('Walmart_Coffee_Listings.db')
cursor = connects.cursor()

### Create Table and Execute that Table

In [ ]:
table = 'CREATE TABLE Walmart_Coffee (id integer primary key, title TEXT, coffee_type TEXT, rating VAR, reviews VAR, seller_name TEXT, thumbnail TEXT, price VAR, weight VAR, weight_formatted_to_gramms VAR )'
cursor.execute(table)

### Insert Data into that Table

In [4]:
insert_query = 'INSERT INTO Walmart_Coffee(title, coffee_type, rating, reviews, seller_name, thumbnail, price, weight, weight_formatted_to_gramms) VALUES(?,?,?,?,?,?,?,?,?)'
file = open('Walmart_Coffee_Listings.csv')
query = csv.reader(file)
cursor.executemany(insert_query, query)

### Have a Look at the First 5 Lines of Our Dataset

In [5]:
select_query = 'SELECT * from Walmart_Coffee LIMIT 10'
for i in cursor.execute(select_query):
    print(i)

(1, 'title', 'coffee_type', 'rating', 'reviews', 'seller_name', 'thumbnail', 'price', 'weight', 'weight_formatted_to_gramms')
(2, 'folgers classic roast ground coffee, 40.3-ounce', 'classic roast', 3.8, 93, 'walmart.com', 'https://i5.walmartimages.com/asr/1fbbd523-8554-4a85-8107-2568a125a6d2.2395bc5a9e08e45dc51e62b268776b65.jpeg?odnHeight=180&odnWidth=180&odnBg=FFFFFF', 13.92, '40.3-ounce', 1142.5)
(3, 'café bustelo, espresso style dark roast ground coffee, vacuum-packed 10 oz. brick', 'espresso,dark roast', 4.7, 914, 'walmart.com', 'https://i5.walmartimages.com/asr/99a53df0-0471-4b63-abe7-d6b8d314ebec_1.bbbb1a66318deade7f16115f60ac8725.jpeg?odnHeight=180&odnWidth=180&odnBg=FFFFFF', 3.76, '10 oz', 283.5)
(4, 'folgers classic roast ground coffee, medium roast coffee, 25.9 ounce canister', 'medium roast,classic roast', 4.4, 740, 'walmart.com', 'https://i5.walmartimages.com/asr/e6aba325-608e-4838-a29d-d63e965a4d85.60c763c73d203bef88f34a464124bcb9.jpeg?odnHeight=180&odnWidth=180&odnBg=FFFF

### Q1: What coffee type has the highest rating?

In [6]:
select_rating = 'SELECT coffee_type, AVG(rating) AS avg_rating from Walmart_Coffee GROUP BY coffee_type ORDER BY avg_rating DESC LIMIT 3'
for i in cursor.execute(select_rating):
    print(i)

('italian', 5.0)
('espresso,mocha', 5.0)
('french,latte,instant coffee', 4.9)


> Top three rating coffee types are **Italian, Espresso(mocha), and French,Latte,Instant Coffee**.

> **Italian** has the highest rating and gets full rating scores of 5.

### Q2: What coffee type has the lowest/highest price?

**· Q2.1 How many distinct coffee types are there in our dataset ？**

> There are **122** coffee types.

In [10]:
count_type = 'SELECT count(DISTINCT(coffee_type)) from Walmart_Coffee'
for i in cursor.execute(count_type):
    print(i[0])

122


**· Q2.2 Which type of coffee has the most data？**

> We have 133 rows of data for the type: **Medium Roast**

> **Medium Roast** is the most popular coffee type

In [8]:
select_type = 'SELECT coffee_type, count(*) from Walmart_Coffee GROUP BY coffee_type ORDER BY count(*) DESC LIMIT 1'
for i in cursor.execute(select_type):
    print(i)

('medium roast', 313)


In [ ]:
select_price_high = 'SELECT coffee_type, AVG(price) AS avg_price from Walmart_Coffee GROUP BY coffee_type ORDER BY avg_price DESC LIMIT 3'
for i in cursor.execute(select_price_high):
    print(i)

> Top three most expensive coffee types are **Italian, Mocha(Light roast), and Arabica** with price 37.99, 35.93, and 26.78.

> **Italian** has the highest price. **Italian and Mocha(Light roast)** are the only two coffee types which have price over 30.

### Q3: What coffee type has the lowest/highest weight?

In [11]:
select_weight_high = 'SELECT coffee_type, AVG(weight_formatted_to_gramms) AS avg_weight from Walmart_Coffee GROUP BY coffee_type ORDER BY avg_weight DESC LIMIT 3'
for i in cursor.execute(select_weight_high):
    print(i)

('mocha,instant coffee', 2381.4)
('caramel,macchiato,iced coffee', 1814.4)
('instant coffee,arabica,dark roast', 1134.0)


In [13]:
select_weight_low = 'SELECT coffee_type, AVG(weight_formatted_to_gramms) AS avg_weight from Walmart_Coffee GROUP BY coffee_type ORDER BY avg_weight LIMIT 5'
for i in cursor.execute(select_weight_low):
    print(i)

('coffee_type', 0.0)
('colombian,instant coffee', 99.2)
('mocha,iced coffee,latte', 99.2)
('instant coffee,light roast', 189.9)
('espresso,instant coffee,dark roast', 199.9)


> Top three heaviest coffee type are **Mocha(instant coffee), Caramel,macchiat(iced coffee), and Instant coffee,arabica,dark roast** with weight 2381.4, 1814.4, and 1134.0 gramms.

> Top four lightiest coffee type are **Colombian(instant coffee), Mocha(iced coffee,latte), Instant coffee(light roast), and Espresso(instant coffee,dark roast)** with weight 99.2, 99.2, 189.9 and 199.9 gramms.